osu.Game/Collections/CollectionManager.cs

```diff
@@ -56,11 +57,14 @@ private void load()
         {
             Collections.CollectionChanged += collectionsChanged;
 
-            if (storage.Exists(database_name))
+            // If a backup file exists, it means the previous write operation didn't complete successfully. Always prefer the backup file in such a case.
+            string filename = storage.Exists(database_backup_name) ? database_backup_name : database_name;
```

Comment:
I think you want to copy the backup back in place before proceeding. Imagine this scenario:

- Something goes wrong, only the backup file is left
- User starts game, then makes a collection change
- `save()` writes temporary file, deletes backup, then attempts to move the "current" file, which doesn't actually exist
- Crash occurs and temporary file is not written to main database location
- All files are now lost

Reply:
Regarding point (3), it will not move the current file if it doesn't exist (there's a `File.Exists()` check around it).

The rest is still possible - backup could then be deleted and the `File.Move()` from the temp locations fails. At which point it only resides in memory.

I'll change this to copy back if it doesn't exist.

Reply:
Yeah, that's what I was implying. Not that it would fail, but it tried to copy and doesn't because of the conditional.